corehq/apps/case_search/temp/fix_child_cases.py

```diff
@@ -0,0 +1,115 @@
+def update_script(case_ids_to_exclude=[]):
+    from corehq.form_processor.models import CommCareCase
+    from corehq.apps.locations.models import SQLLocation
+    from casexml.apps.case.mock import CaseBlock
+    from corehq.apps.hqcase.utils import submit_case_blocks
+    import os
+
```

Comment:
A few nitpicky things, and a gotcha:

1. Nit: Conventionally, scripts are marked executable (`$ chmod +x myscript`) and start with a "shebang" to tell Linux/*NIX how to execute them. In the case of Python 3 scripts, that's `#!/usr/bin/env python3`, which will call whichever Python interpreter is default for the environment (i.e. it respects virtual environments).
2. Nit: No need for local imports. You can use normal module-level imports.
3. Gotcha: Don't set the default value of a parameter to a mutable data type. [Unexpected things can happen](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).

```suggestion
#!/usr/bin/env python3

from corehq.form_processor.models import CommCareCase
from corehq.apps.locations.models import SQLLocation
from casexml.apps.case.mock import CaseBlock
from corehq.apps.hqcase.utils import submit_case_blocks
import os

def update_script(case_ids_to_exclude=None):
    if case_ids_to_exclude is None:
        case_ids_to_exclude = set()
```
(When checking whether something is a member of a group, a set is faster than a list or a tuple.)

Then at the end of a Python script, you'll usually call the top-level function like so:
```python
if __name__ == '__main__':
    # (Parse command line arguments here.)
    # (Maybe get `case_ids_to_exclude`.)
    update_script()
```

Reply:
Just an FYI: My plan is to copy paste this in the django shell :smile: